src/blocks/review-comparison/edit.js

```diff
@@ -0,0 +1,240 @@
+/** @jsx jsx */
+
+/**
+ * External dependencies
+ */
+import {
+	css,
+	jsx
+} from '@emotion/react';
+
+import classnames from 'classnames';
+
+import getSymbolFromCurrency from 'currency-symbol-map';
+
+/**
+  * WordPress dependencies.
+  */
+import { __ } from '@wordpress/i18n';
+
+import apiFetch from '@wordpress/api-fetch';
+
+import {
+	Fragment,
+	useEffect,
+	useState
+} from '@wordpress/element';
+
+/**
+  * Internal dependencies
+  */
+import defaultAttributes from './attributes.js';
+import Placeholder from './placeholder.js';
+import Controls from './controls.js';
+import Inspector from './inspector.js';
+import { blockInit } from '../../helpers/block-utility.js';
+import defaultReviewAttributes from '../review/attributes.js';
+import {
+	StarFilled,
+	StarHalf
+} from '../../helpers/icons.js';
+
+let tableImages = [];
+let tableName = [];
+let tablePrice = [];
+let tableRating = [];
+let tableDescription = [];
+let tableStatistics = [];
+let tableLinks = [];
+
+const Edit = ({
+	attributes,
+	setAttributes,
+	className,
+	clientId
+}) => {
+	useEffect( () => {
+		const unsubscribe = blockInit( clientId, defaultAttributes );
+		return () => unsubscribe();
+	}, [ attributes.id ]);
+
+	useEffect( () => {
+		( async() => {
+			try {
+				setStatus( 'loading' );
+				const data = await apiFetch({ path: 'themeisle-gutenberg-blocks/v1/filter_blocks?block=themeisle-blocks/review' });
+				setData( data );
+				setStatus( 'loaded' );
+				setEditing( ! Boolean( attributes.reviews.length ) );
+			} catch ( error ) {
+				setStatus( 'error' );
+			}
+		})();
+	}, []);
+
+	const [ data, setData ] = useState([]);
+	const [ status, setStatus ] = useState( 'loading' );
+	const [ isEditing, setEditing ] = useState( true );
+
+	const styles = css`
+		.wp-block-themeisle-blocks-review-comparison__buttons span {
+			background: ${ attributes.buttonColor } !important;
+			color: ${ attributes.buttonText } !important;
+		}
+	`;
+
+	useEffect( () => {
+		tableImages = [];
+		tableName = [];
+		tablePrice = [];
+		tableRating = [];
+		tableDescription = [];
+		tableStatistics = [];
+		tableLinks = [];
+
+		if ( ! Boolean( data.length ) ) {
+			return;
+		}
+
+		attributes.reviews.map( value => {
```

Comment:
Let's avoid using `map` as a for-loop. For this kind of thing, we have `forEach`. 

`Map` is defined as a transformer. You have a list of elements of type A and applying a function will transform them into a list of types B.

Reply:
Yea, you're right. I have this habit of using map in react JSX so I carried it forward to code.